Heiko Code Review 3

CodeLtDave opened the pull request.

CodeLtDave committed:
aktueller Stand kinda working

asda

CodeLtDave committed:
Revert "aktueller Stand kinda working"

This reverts commit 78cfb44d6532ee927103c8c98c154a441fe86af8.

CodeLtDave committed:
MakeStatic files init

Commit:
Almost working

Commit:
kleiner Schubser

CodeLtDave committed:
Working Rewrite but other errors

CodeLtDave committed:
Revert "Working Rewrite but other errors"

This reverts commit 9a56081415a75f1029d56aaea3674a292a739c18.

Commit:
right before change working

Commit:
working

Commit:
HEUREKA

CodeLtDave committed:
fix public modifier disappearing

leave the remeaining modifiers instead of replacing them

CodeLtDave committed:
beautify

CodeLtDave committed:
cleanup

CodeLtDave committed:
Revert "cleanup"

This reverts commit 8c021e42514c000b4628f2b5b16a949f11b9ed18.

Commit:
InvocationUpdate added but not working

CodeLtDave committed:
Chage Invocations, work in progress

Commit:
Working Invocation change in the same class

Commit:
Refactoring now working in different classes

MethodInvocations will now also be changed in different classes within the same project. In other projects its not working properly.

Commit:
Bugfix

Bugfix for Refactoring of more than one method in the same class that individually contain a Methodinvocation

CodeLtDave committed:
Merge branch '_MichaelNEW' of https://github.com/CodeLtDave/eclipse.jdt.ui into _MichaelNEW

Commit:
Bugfix

Fixed Bug when refactoring a method with another return type than void. Now the Refactoring works for all return types.

Commit:
Changed MethodInvocations don't overwrite arguments anymore

Commit:
First MakeStaticRefactoring Test implemented

Commit:
First Test Working

CodeLtDave committed:
better code structure and cleanup

Commit:
More Tests added

Commit:
Additional Tests and Bugfix for Refactoring

If MethodDeclaration is modified before MethodInvocations, the Refactoring will fail

Commit:
ThrowException Test working

CodeLtDave committed:
added ChangeManager which handels adding edits

Before there could be multiple CompilationUnitChanges for one CompilationUnit. Now you can use

addEditToChangeManager(TextEdit, ICompilationUnit)

so that this is automatically take care off. If there is a CompilationUnitChange already its pulled from ChangeManager and edited instead of having multiple CompilationUnitChanges on one ICompilationUnit.

CodeLtDave committed:
cleanup

CodeLtDave committed:
small fix

CodeLtDave committed:
small refactorings

Commit:
new tests

new tests for every type as parameter and return type

Commit:
new test for this-usage in MethodDeclaration

Commit:
Additional Tests for multiple refactored files

CodeLtDave committed:
Improve method invocation handling in MakeStaticRefactoring

Enhance the MakeStaticRefactoring class to better handle method
invocations in both instance and static contexts. Adjustments
include passing 'this' as the new argument for invocations within
instance methods and passing the instance variable for invocations
in other classes or static methods.

CodeLtDave committed:
Merge branch '_MichaelNEW' of https://github.com/CodeLtDave/eclipse.jdt.ui into _MichaelNEW

Commit:
small change in test

Commit:
Additional test for this + InnerClass

Commit:
small fix for test

CodeLtDave committed:
Implement handling of instance usages in method refactoring now with a CHECK

This is not finished yet, there are 2 tests failing but at least no errors anymore

In this commit, we introduced the fHasInstanceUsages flag to track if a method being refactored has any instance variables or methods. If fHasInstanceUsages is true, the method will be modified to accept an additional parameter representing the instance of the class.

Key changes include:

Implementing an ASTVisitor to visit SimpleName nodes within the method body and check for instance variables or methods
If an instance variable is found, the code replaces the variable with an appropriate FieldAccess expression
If an instance method is found, the code replaces the method invocation with an appropriate MethodInvocation expression

This update improves the refactoring process by correctly handling methods with instance usages, enabling a more accurate static method conversion.

Commit:
Beautify tests and proper suggestion of refactoring

Based on the specific selection of the user the refactoring will now only be suggested if the user selects a method.

Commit:
Additional functionality and adjustments

Additional checks for Refactoring in checkInitialConditions. Editor Selection for Refactoring is now more precisely.
Adjustments for tests for more accurate selection.
Refactoring can now be executed from outline window.
Message Constants included for better Exception and RefactoringStatus handling.

Commit:
Adjustments to tests

Commit:
New Tests and handling for SuperMethodInvocations

Commit:
Little fix for tests that expect error

Commit:
Implemented ASTMatcher for Tests

The method matchASTs is more precise and ignores eventual whitespaces in the files. With the method matchFiles it is easier to understand where the files have differences.  Both methods can be used for testing

Commit:
Check if parameter with same name already exists

if Method has instance usage a new parameter is introduced to the method. If this parameter has the same name as an already existing parameter then an error will be thrown.

Commit:
Check if duplicate method signature exists

when refactoring a method with instance usage, a new parameter is introduced. After the refactoring the method could have the same signature as an already existing method. This check throws an Error in this case.

Commit:
Bugfix Overlapping TextEdits when status has error

Commit:
Multiple Updates

- Better Handling of SuperMethodInvocations.
- Deletion of @Override Annotation.
- Better Handling of StatusErrors -> immediate return after error

Commit:
ShortCut implemented: Alt+Shift+K

Commit:
Bugfix for box name on Refactoring errors

Commit:
MakeStatic can now be used from upper Refactoring menu

Commit:
Cleanup

Commit:
Check if selected method is already static + Bugfix

- new test for this case
- new error message
- fixed a bug where an "Overlapping text edits" error was thrown when pressing back in WIzard after an error occured and then again pressing continue to continue the refactoring.
- Changed a line, so that MakeStatic error message and not IntroduceIndirection error  message will be thrown in MakeStaticAction

Commit:
Adjusted Handling of SuperMethodInvocations and methods that are overriden in TypeHierarchy

Commit:
Code Cleanup

- deleted method getCachedCURewrite()
- more comments for better understanding
- fMethodDeclaration is not initialized in checkInitialConditions() and not in findMethodDeclaration()
- findMethodDeclaration() is now called checkForInstanceUsagesInMethodDeclaration() and is only for this use.

Commit:
New Tests for special cases

- Generic declaration
- Non Static Inner Class

CodeLtDave committed:
tests working

CodeLtDave committed:
Merge branch '_DavidNEW' into _MichaelNEW

Commit:
Additional tests and handling

Tests for following cases:
- Usage of Generics
- refactoring a method should not add additional parameter in case of field access whenparameter already exists
- Selected method is overriding a method of parent type and has no parameter -> refactoring would change method so that it hides parent type

Also included handling for last case

Commit:
Additional Tests

new tests for following cases:
- Field access in anonymous class of a field that belongs to the outer class
- The same case but in a lambda and not in an anonymous class

CodeLtDave committed:
removed findMethodDeclaration and added functionality to new inner class"ChangeInstanceUsagesInMethodBody"

Commit:
New Tests

- Many new tests for various use cases

Commit:
Better handling of duplicate method signature

Before method were only compared with number of parameters and  method name. Now also the parameter types are compared

Commit:
Better handling for duplicate parameterName

Commit:
Handling for static method in lower hierarchy

- handling added if method with same signature in lower hierarchy is static -> refactoring only throws error if method in lower hierarchy overrides selected method, but a static method can not override.
- Bug fixed for test

Commit:
Adjustments to tests, Little change to input page, Bugifx

- Fixed a bug where methods that override parent method and have instance usage couldnt be refactored
- Adjustments to tests so they work properly
- Deleted not needed tests
- Little Change to Input Page and Wizard

Commit:
Added warning when refactoring method that overrides

- Changed dialog window size back to default

Commit:
implemented check for recursive method

- User gets error message when refactoring a recursive method.
- This behavior does not apply for SuperMethodinvocations. Methods that override the parent method and call it with super can still be refactored.

Commit:
Added functionality for updating javadocs

Commit:
Adjustment to ExtraTests

Commit:
Added functionality for Generics

- Adjustments to tests
- methods with generics can now be refactored but only if they are not already parametrized with type parameters

Commit:
Updated Generic functionality

- methods that use less generic parameters than the amount of type parameters of the class will now be refactored correctly. Only the exact amount of needed parameters will be added to the type parameter list of the method.
- New Tests for special cases with generics added

Commit:
bug fix if method has array as parameter

Commit:
New Test

Commit:
Several bugfixes for generics functionality

- additional test

Commit:
new Generics functionality

- added functionality for already parametrized methods
- added functionality for bounds of typeParameters

Commit:
little fix for test

- parametrized methods can be refactored now, so test doesnt need to check for error

Commit:
check for WildCardTypes as bounds implemented

- additional Tests and check for wildCardTypes as bounds of typeparam of class. (T extends List<?>)

Commit:
Bugfix and adjustment to tests

- Refactoring is now possible if a method does not override the parent type method but has the same name as the parent type method

Commit:
Bugfix for generics in JavaDoc

- now braces <T> are also added to JavaDoc when adding type parameters

Commit:
check for LambdaExpression and MethodReference

- if Lambda is detected as MethodInvocation the refactoring is cancelled
- bugfix for invoking refactoring on MethodInvocation that is in different class than MethodDeclaration
- code cleanup

Commit:
Functionality for VarArgs included

- new parameter for instance access is now inserted as first parameter of method, so that varargs are always the last parameter
- changed every test that adds new parameter to new behavior

Commit:
Cleanup

- extracted methods from method modifyMethodDeclaration() for more clarity

Commit:
little change

CodeLtDave committed:
fix: hasInstanceUsage and InnerClassCreatation

fix for wrongfully recognised hasInstanceUsages and creating anonymous inner instance classes

Commit:
Bugfix for generics

- updateTypeParamList now also includes typeParams of parametrized methodParams

Commit:
small changes to some tests

Commit:
changes to Tests

CodeLtDave committed:
unfinished state before extracting handleInstanceVariable

Commit:
merging of test classes and changes for MethodReferences

- Changed behaviour for MethodReferences and SuperMethodReferences
- LambdaExpressions are now allowed for Refactoring
- Merging of test classes

Commit:
New tests

- It is not possible anymore to refactor a method that uses explicit super method invocations in the methodDeclaration body
- New warning for explicit access of fields from the parent type with the super keyword
-

Commit:
Comparing of methods is now more precise

Commit:
new test for foncatenated field access

CodeLtDave committed:
extend testConcatenatedFieldAccessAndQualifiedNames

CodeLtDave committed:
fixed testConcatenatedFieldAccessAndQualifiedNames

CodeLtDave committed:
fixes for modifyMethodInvocation and concatinated FieldAccess/QualifiedName

CodeLtDave committed:
fixes for modifyMethodInvocation

CodeLtDave committed:
testsWorking, fixed behavior for InnerFieldAccessInAnonymousClass

Commit:
Deleted output files for tests with error...

-  check if source is available for selected method
- New test for this case

CodeLtDave committed:
checks for anonymous/inner classes

CodeLtDave committed:
Merge branch '_MichaelNEW' of https://github.com/CodeLtDave/eclipse.jdt.ui into _MichaelNEW

Commit:
right output files deleted and others inserted again

CodeLtDave committed:
fixed test22?

CodeLtDave committed:
Letzter SCHUBSSER

Commit:
Merging of tests and new test for nested classes

Commit:
Test merging completed

CodeLtDave committed:
fixed testNested

CodeLtDave committed:
comment refactorings to modifyMethodDeclaration

CodeLtDave committed:
refactorings, broke testInheritance3

CodeLtDave committed:
structure fix to checkNode

Commit:
fixed bug for testInheritance3

Commit:
error messages now better to understand

Commit:
deleted obsolete test testNoAdditionalParameter2

Commit:
Put all "then" statements into a block for readability. Autoformated code

CodeLtDave committed:
Revert accidental change in file

Commit:
improved readability for test class

CodeLtDave committed:
Move MakeStaticAction to org.eclipse.jdt.ui\ui\org\eclipse\jdt\ui\actions

CodeLtDave committed:
Move InputPage and Wizard

CodeLtDave committed:
Move MakeStaticRefactoring.java

CodeLtDave committed:
Revert "Move MakeStaticRefactoring.java"

This reverts commit 6005e5706ab3510f5f5dcc03988d85f3a1fe779c.

Commit:
changes for better understanding

- changed dialog title
- outsourced strings to localized variable

CodeLtDave committed:
Move MakeStaticRefactoring.java

CodeLtDave committed:
Merge branch '_DavidNEW' of https://github.com/CodeLtDave/eclipse.jdt.ui into _DavidNEW

CodeLtDave committed:
MakeStaticWizard removed unused imports

CodeLtDave committed:
Added "&k" symbol for acessing the action via keypress

CodeLtDave committed:
adjusted copyright notice MakeStaticAction

CodeLtDave committed:
Merge branch '_DavidNEW' of https://github.com/CodeLtDave/eclipse.jdt.ui into _DavidNEW

CodeLtDave committed:
Modified all Copyright notices

CodeLtDave committed:
changed AST version for parser to IASTSharedValues.SHARED_AST_LEVEL

CodeLtDave committed:
java docs edit, "if...return" restructure

Commit:
Improved code readability for inner class ChangeInstanceUsagesInMethodBody

Commit:
changed inner class ChangeInstanceUsagesInMethodBody to a standalone class

Commit:
deleted unnecessary line of code

Commit:
CleanUp MakeStaticInputPage

Commit:
new names for variables in InputPage

Commit:
small fix to Exception and deleted unnecessary null check

Commit:
JavaDocs for MakeStaticWizard

Commit:
name change for ChangeInstanceUsageInMethodBody

CodeLtDave committed:
create package for MakeStaticRefactoring.java, class split preperation

Commit:
JavaDocs for InstanceUsageRewriter

Commit:
JavaDiocs for MakeStaticRefactoring

Commit:
JavaDocs for fields in MakeStaticRefactoring

Commit:
Outsourced inner class MethodReferenceFinder and simplified its methods

Commit:
JavaDocs for MakeStaticRefactoring methods

CodeLtDave committed:
init ContextCalculator

Commit:
Merged methods isOverridden and is Overriding

Commit:
added field fStatus and changed all usages of status

Commit:
Copyright comment for classes

CodeLtDave committed:
progress

Commit:
improved readability for method modifyMethodInvocation

Commit:
tests for javaDocs

CodeLtDave committed:
unfinished

Commit:
improved readability

Commit:
almost working

Commit:
added general initial condition checks

Commit:
bugfix

Commit:
deleted throw

we make a daily now

Commit:
bugfix for selection of MethodInvocation before starting refactoring

Commit:
all tests working but bad code style

CodeLtDave committed:
logic fix and java docs

CodeLtDave committed:
Unified code paths for TextSelection and IMethod inputs

Unified code paths by calculating IMethod for text selection input, reducing code complexity and improving maintainability

CodeLtDave committed:
refactorings and java docs

CodeLtDave committed:
Merge branch '_DavidNEW' into _MichaelNEW

CodeLtDave committed:
merge JavaDocs and fStatus field

Commit:
small refactorings

and changing back method isOverriding to its initial state because the combination of isOverridden and isOverriding makes no sense anymore.

Commit:
small refactorings and created new class FinalConditionsChecker

Commit:
Outsourced checks

Every check from method checkFinalConditions is now checked from the class FinalConditionsChecker.

Only checks from classes MethodReferenceFinder and InstanceUsageRewriter are not outsourced yet.

Commit:
outsourced checks of MethodReferenceFinder to FinalConditionsChecker

Commit:
changed InitialConditionsChecker methods to static

Commit:
JavaDocs for InititalConditionsChecker

Commit:
fixed copyright message

Commit:
JavaDocs and copyright note for FinalConditionsChecker

Commit:
additional copyright notes

CodeLtDave committed:
introduce concept of getOrCompute in ContextCalculator

CodeLtDave committed:
Rethrow JavaModelException in business logic layer

This commit addresses an issue where a JavaModelException was being caught
and handled within a class that should only be responsible for business logic.

Instead of handling the exception and logging it within the method, the exception
is now thrown back to the caller. This allows the caller to decide how best to handle
the error, whether that be rethrowing the exception, returning an error, or presenting
the error to the user in a UI dialog.

This change also prevents potential NullPointerExceptions that could occur when the
method returns a null value. These NullPointerExceptions could cause the refactoring
process to fail and present the user with both console logging and a popup error message.

The changes made align the class more closely with the principle of separation of concerns,
keeping business logic and UI interaction separate.

CodeLtDave committed:
Rethrow JavaModelException in business logic layer 2

Commit:
Init ChangeCalculator

CodeLtDave committed:
missing import JavaModelException

CodeLtDave committed:
Restructure makestatic package for improved isolation

Michael5601 committed:
changed name of method modifyMethodDeclaration

Michael5601 committed:
addInstanceAsParamIfUsed is now executed before the two checks beneath it

Michael5601 committed:
deleted duplicate description in javaDocs of method getName

Michael5601 committed:
JavaDocs for MakeStaticRefactoring

Michael5601 committed:
JavaDocs and removing unnecessary fields in ChangeCalculator

Michael5601 committed:
InititalConditionsChecker is now instantiated

all checks in checkInitialConditions are now more concise and easier to read. The checker methods are instance methods now and return a boolean value instead of the RefactoringStatus. The InititalConditionsChecker is instantiated with a RefactoringStatus object and uses it in every checker method to determine which boolean value needs to be returned.

Michael5601 committed:
bugfix and updated @return tag in JavaDocs

Michael5601 committed:
updated description for @throws tag in JavaDocs

Michael5601 committed:
refactor code in ChangeCalculator.

Michael5601 committed:
cleaner solution for code in generateUniqueParameterName

while (true) is changed to do while

Michael5601 committed:
changed variable name of counter to incrementingSuffix

Michael5601 committed:
reduced instance variables

Deleted instance variables in MakeStaticRefactoring and replaced the usage with getOrCompute method invocations.

Michael5601 committed:
changed the direct usage of an instance variable to the matching get method invocation

Michael5601 committed:
changed name of index i to parameterNumber

Michael5601 committed:
description for exception

Michael5601 committed:
moved final condition checks from MakeStaticRefactoring into ChangeCalculator

CodeLtDave committed:
remove try catch

CodeLtDave committed:
Merge branch '_MichaelNEW' of https://github.com/CodeLtDave/eclipse.jdt.ui into _MichaelNEW

CodeLtDave committed:
removed try catch blocks

CodeLtDave committed:
removed fTargetMethodDeclaration field

Michael5601 committed:
modifications to MethodDeclaration are capsulated in modifyMethodDeclaration

- Changecalculator has new field fInstanceUsageRewriter and field fTargetMethodhasInstanceUsage is delted instead. Everywhere fTargetMethodhasInstanceUsage was used the getter getTargetMethodhasInstanceUsage from InstanceUsageRewriter is used

CodeLtDave committed:
Refactor generateUniqueParameterName method

    - Improved variable naming for better readability.
    - Removed redundant null check on method parameters.
    - Streamlined loop logic to improve performance.
    - Created a separate method `parameterNameExists` to handle the concern of checking parameter name existence.

CodeLtDave committed:
Merge branch '_MichaelNEW' of https://github.com/CodeLtDave/eclipse.jdt.ui into _MichaelNEW

Michael5601 committed:
JavaDocs for modifyMethodDeclaration

Michael5601 committed:
getChanges is now getOrComputeChanges

getOrComputeChanges checks if changes are present and if not modifyMethodDeclaration is performed before returning allchanges

Michael5601 committed:
addition to javadocs

CodeLtDave committed:
added Copyright JavaDoc to ChangeCalculator

CodeLtDave committed:
removed outdated ToDo

CodeLtDave committed:
Merge branch '_MichaelNEW' of https://github.com/CodeLtDave/eclipse.jdt.ui into _MichaelNEW

CodeLtDave committed:
added fallback logic to ContextCalculator for Exceptions

CodeLtDave committed:
added fallback logic to ContextCalculator for Exceptions 2

CodeLtDave committed:
small fix

Michael5601 committed:
The RefactoringStatus is now fully managed by FinalConditionsChecker and InitialConditionsChecker

- all checks of checkFinalConditions take place in FinalConditionsChecker now. No RefactoringStatus is returned by any method except checkFInalConditions and checkInitialConditions

Michael5601 committed:
Merge branch '_MichaelNEW' of https://github.com/CodeLtDave/eclipse.jdt.ui into _MichaelNEW

CodeLtDave committed:
introduced progess monitor

Michael5601 committed:
JavaDocs for MethodReferenceFinder

CodeLtDave committed:
Merge branch '_MichaelNEW' of https://github.com/CodeLtDave/eclipse.jdt.ui into _MichaelNEW

CodeLtDave committed:
auto format

CodeLtDave committed:
Merge branch '_MichaelNEW' of https://github.com/CodeLtDave/eclipse.jdt.ui into _MichaelNEW

CodeLtDave committed:
Merge branch 'master' into _MichaelNEW

Michael5601 committed:
JavaDocs for InstanceUsageRewriter

Michael5601 committed:
JavaDocs for FinalConditionsChecker

Michael5601 committed:
JavaDocs for ChangeCalculator

CodeLtDave committed:
adjustements to fork sync

CodeLtDave committed:
Convert IProgressMonitor to SubMonitor

CodeLtDave committed:
add checkCancel for SubMonitor

Michael5601 committed:
bugfix for NLS Message and deleted MakeStaticRefactoring_parameter_name_already_used because it is not used anymore

Michael5601 committed:
Merge branch '_MichaelNEW' of https://github.com/CodeLtDave/eclipse.jdt.ui into _MichaelNEW

Michael5601 committed:
Fixed bug where user gets every error message shown twice

Michael5601 committed:
correct name when undoing changes with Ctrl+Z

HeikoKlare commented on org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/code/MakeStaticRefactoring.java:
```diff
@@ -0,0 +1,224 @@
+				return fStatus;
+			}
+		}
+		progress.checkCanceled();
```
`split` already calls `checkCanceled`, so you can remove the calls to `checkCanceled`.

Michael5601 replied:
Done. 187f5f133f

HeikoKlare commented on org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/code/MakeStaticRefactoring.java:
```diff
@@ -0,0 +1,224 @@
+		//Find and modify MethodInvocations
+		fChangeCalculator.handleMethodInvocations(progress.slice(1), fContextCalculator.getOrComputeTargetIMethodBinding());
+		progress.checkCanceled();
+		progress.split(1);
```
You only seem to fill 7 of 8 parts in the progress.

Michael5601 replied:
Done. 187f5f133f

HeikoKlare commented on org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/code/makestatic/ContextCalculator.java:
```diff
@@ -0,0 +1,253 @@
+
+	private ASTNode fSelectionASTNode;
+
+	/**
+	 * The {@code IMethodBinding} object representing the binding of the refactored method.
+	 */
+	private IMethodBinding fTargetIMethodBinding;
+
+	/**
+	 * The {@code IMethod} object representing the selected method on which the refactoring should
+	 * be performed.
+	 */
+	private IMethod fTargetIMethod;
```
Is there a reason why two of the fields are documented while the others are not?

Michael5601 replied:
Normally we did not document private fields but in the process of creating the ContextCalculator and deleting some redundant fields of MakeStaticRefactoring I transfered the already existing javaDocs of some fields from MakeStaticRefactoring to ContextCalculator. I will also document the other fields because it could benefit the next person that wants to debug our code.

Michael5601 replied:
Done. 85eeeac64e

HeikoKlare commented on org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/code/makestatic/ContextCalculator.java:
```diff
@@ -0,0 +1,253 @@
+			return;
+		}
+		fTargetMethodDeclaration= getMethodDeclarationFromIMethod(fTargetIMethod, fTargetCompilationUnit);
+		fTargetIMethodBinding= fTargetMethodDeclaration.resolveBinding();
```
You have to dive deeo into the code to see that `fTargetIMethod` will never be `null` because of calling `calculateTargetCompilationUnit()` before. It would be even better if all the `calculate...` methods first call the `getOrCompute...` methods for the variables they require and store the values in local variables that they access later on. Then you can check all the values for not being `null` and safely use them afterwards. Then it is (and will always be even upon changes to the class) guaranteed that those values are never `null` when using them.

Here, e.g., it would look like this:
```suggestion
		IMethod targetIMethod = getOrComputeTargetIMethod();
		CompilationUnit targetCompilationUnit = getOrComputeTargetCompilationUnit();
		if (targetIMethod == null || targetCompilationUnit == null) {
			return;
		}
		fTargetMethodDeclaration= getMethodDeclarationFromIMethod(targetIMethod, targetCompilationUnit);
		fTargetIMethodBinding= fTargetMethodDeclaration.resolveBinding();
```

CodeLtDave replied:
Done.
2a75304fb32d8daeccb0ad773ace6e755e9b6e43

HeikoKlare commented on org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/code/makestatic/InitialConditionsChecker.java:
```diff
@@ -0,0 +1,203 @@
+
+	/**
+	 * Checks if the start position of a text selection is valid. A Selection is valid if the offset
+	 * and lengt are greater zero.
```
```suggestion
	 * and length are greater zero.
```

HeikoKlare commented on org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/code/makestatic/ChangeCalculator.java:
```diff
@@ -0,0 +1,546 @@
+		if (classTypeParameters.length != 0) {
+			SimpleType simpleType= fTargetMethodDeclarationAST.newSimpleType(fTargetMethodDeclarationAST.newName(className));
+			ParameterizedType parameterizedType= fTargetMethodDeclarationAST.newParameterizedType(simpleType);
+			for (int i= 0; i < classTypeParameters.length; i++) {
```
That's a matter of personal preference as many people accept `i` and other letters as a valid loop variable name, but I prefer to also have speaking names there (e.g. `parameterNumber` in this case) to make the code easier to read.

Michael5601 replied:
Done. 21b4bcb6a2

HeikoKlare commented on org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/code/makestatic/ChangeCalculator.java:
```diff
@@ -0,0 +1,546 @@
+			allTextEdits= new MultiTextEdit();
+		}
+		allTextEdits.addChild(editToAdd);
+		String changeName= "Change in " + iCompilationUnit.getElementName(); //$NON-NLS-1$
```
This should be captured in an NLS to allow localization for different languages.

Michael5601 replied:
Done. 922a1b1db3

HeikoKlare reviewed (commented):
The code now looks really good from a quality perspective. I have only some minor and rather nitpicky comments left.

With respect to functionality, we will also have a review and validation for the official Eclipse pull request, so I won't evaluate functionality beyond what is covered by the tests anyhow now.

Michael5601 committed:
Update org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/code/makestatic/InitialConditionsChecker.java

Co-authored-by: Heiko Klare <dev@heiko.klare.info>

Michael5601 committed:
String captured in NLS for future localization

Michael5601 committed:
new speaking name for loop variable i

Michael5601 committed:
JavaDocs for fields in ContextCalculator

Michael5601 committed:
deleted explicit calls of progress.checkCanceled because progress.split already calls it

- increment of split number in checkFinalConditions so that it fits the 8 splits in the submonitor

CodeLtDave committed:
Apply suggestions from code review

Co-authored-by: Heiko Klare <dev@heiko.klare.info>

CodeLtDave committed:
Merge branch 'eclipse-jdt:master' into _MichaelNEW

CodeLtDave committed:
Refactored calculate Methods to store values as local variables

https://github.com/CodeLtDave/eclipse.jdt.ui/pull/5/files/e7ee4ca4aa1d1dba44536d9910493b1ee429e70f#r1276408436
